### ISSUE TITLE
Possibly incorrectly typed objectEntries not returning an array type

### ISSUE THREAD

**Issue by jalooc** [garronej/tsafe#17]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

I think that `objectEntries` is incorrectly typed - it currently returns a union of possible entries, while it should return **an array** of union of possible entries in order to match the `Object.entries()` functionality. (note: the actual js implementation is correct though). Also, I think it incorrectly adds `undefined` to the union of possible entries if at least one of values is optional, as `{ [Key in keyof O]: [Key, O[Key]]; }[keyof O]` doesn't seem to logically allow `undefined` (nor the resulting array of entries from `Object.entries()` returns `undefined` in such case).

---

Sample code:
```ts
const a: { a?: number, b: number, c: number } = ...
const e = objectEntries(a)
```
type of `e`:
```ts
const e: ["a", (number | undefined)] | ["b", number] | ["c", number] | undefined
```
type of `e` after the fix presented below:
```ts
 (["a", (number | undefined)] | ["b", number] | ["c", number])[]
```

---

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/tsafe/objectEntries.d.ts b/node_modules/tsafe/objectEntries.d.ts
index 1f95606..b941470 100644
--- a/node_modules/tsafe/objectEntries.d.ts
+++ b/node_modules/tsafe/objectEntries.d.ts
@@ -1,8 +1,9 @@
 /** https://docs.tsafe.dev/objectentries */
-export declare function objectEntries<O extends Record<string, any>>(o: O): {
+export declare function objectEntries<O extends Record<string, any>>(o: O): NonNullable<{
     [Key in keyof O]: [Key, O[Key]];
-}[keyof O];
+}[keyof O]>[];
+
 /** Return type of objectEntries https://docs.tsafe.dev/objectentries */
-export declare type ObjectEntries<O extends Record<string, any>> = {
+export declare type ObjectEntries<O extends Record<string, any>> = NonNullable<{
     [Key in keyof O]: [Key, O[Key]];
-}[keyof O];
+}[keyof O]>[];
```

---

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by garronej** [comment]
Thank you very much!  
I am quite embarassed to have made such a big mistake.  

I incorporated your change, just used `Exclude<..., undefined>` instead of `NonNullable<...>` since it gives a suggestion more readable.  

![image](https://user-images.githubusercontent.com/6702424/194781761-398c77d7-3c66-4381-97f0-19ed5a337f0c.png)

It's available in tsafe v1.1.2.

**Comment by jalooc** [comment]
Glad I could help :)
